File: src/main/java/com/jafarov/quiz/entity/Question.java
package com.jafarov.quiz.entity;

import jakarta.persistence.*;

@Entity
public class Question {

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;

        private String title;

        @ManyToOne
        @JoinColumn(name = "quiz_id")
        private QuizEntity quiz;

        public Question() {
        }

        public Long getId() {
            return id;
        }

        public void setId(Long id) {
            this.id = id;
        }

        public String getTitle() {
            return title;
        }

        public void setTitle(String title) {
            this.title = title;
        }

        public QuizEntity getQuiz() {
            return quiz;
        }

        public void setQuiz(QuizEntity quiz) {
            this.quiz = quiz;
        }
    }

File: src/main/java/com/jafarov/quiz/admin/entity/Question.kt
package com.jafarov.quiz.admin.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.Table
import org.hibernate.annotations.Nationalized


@Table(name = "questions")
@Entity
open class Question @JvmOverloads constructor(

    @Column(name = "question", nullable = false)
    @Nationalized
    open var question: String? = null,

    @Column(name = "is_active", nullable = false)
    open var isActive: Boolean = false,

    @Column(name = "topic_id")
    open var topicId: Long? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "topic_id", insertable = false, updatable = false)
    open var topic: Topic? = null,

    @OneToMany(mappedBy = "question", fetch = FetchType.LAZY)
    open var answers: MutableSet<Answer>? = null

):BaseEntity()